examples/lump-mpi.py

```diff
@@ -0,0 +1,245 @@
+            mydt = get_inviscid_timestep(discr=discr, q=state,
+                                         cfl=current_cfl, eos=eos)
+        if (current_t + mydt) > t_final:
+            mydt = t_final - current_t
+        return mydt
+
+    def my_rhs(t, state):
+        return inviscid_operator(discr, q=state, t=t, boundaries=boundaries, eos=eos)
+
+    def my_checkpoint(step, t, dt, state):
+        # This stuff should be done by any/all checkpoint
+        current_t = t
+        current_step = step
+        current_state = state
+        current_dt = dt
+        checkpoint_status = 0
+
+        if constant_cfl is False:
+            current_cfl = get_inviscid_cfl(discr=discr, q=state,
+                                           eos=eos, dt=current_dt)
+
+        # The rest of this checkpoint routine is customization
+        do_status = check_step(step=step, n=nstatus)
+        do_viz = check_step(step=step, n=nviz)
+        if do_viz is False and do_status is False:
+            return checkpoint_status
+
+        dv = eos(q=current_state)
+        expected_state = initializer(t=current_t, x_vec=nodes, eos=eos)
+
+        if do_status is True:
+            statusmesg = make_status_message(t=current_t, step=current_step,
+                                             dt=current_dt,
+                                             cfl=current_cfl, dv=dv)
+            max_errors = compare_states(state, expected_state)
+            statusmesg += f"\n------   Err({max_errors})"
+            if rank == 0:
+                logger.info(statusmesg)
+
+            maxerr = np.max(max_errors)
+            if maxerr > exittol:
+                logger.error("Solution failed to follow expected result.")
+                checkpoint_status = 1
+
+        if do_viz:
+            checkpoint_t = current_t
+            visfilename = make_visfile_name(basename=casename, rank=rank,
+                                            step=step, t=checkpoint_t)
+            pvisfilename = visfilename
+            visnamelist = []
+            if rank == 0:
+                pvisfilename = make_parallel_visfile_name(basename=casename,
+                                                          step=step,
+                                                          t=checkpoint_t)
+                visnamelist.append(pvisfilename)
+                visnamelist += [make_visfile_name(basename=casename, rank=i,
+                                                  step=step, t=checkpoint_t)
+                                for i in range(num_parts)]
+
+            io_fields = make_io_fields(dim, state, dv, eos)
+            io_fields.append(("exact_soln", expected_state))
+            result_resid = state - expected_state
+            io_fields.append(("residual", result_resid))
+            visualizer.write_vtk_file(visfilename, io_fields,
+                                      par_namelist=visnamelist,
+                                      overwrite=True)
```

Comment:
How come CI passed on (in #13) without pointing meshmode at https://github.com/inducer/meshmode/pull/20 in `requirements.txt`?

Reply:
The run_examples.sh improperly does not return an error code when one of the examples fails, so the failing example that need that viz *do return a proper error code* but [this script ](https://github.com/illinois-ceesd/mirgecom/blob/c3f6f2b26c607b8fc065b7e8a115ed17d6fe1418/examples/run_examples.sh#L1)doesn't capture it (yet).

Reply:
#45